core/src/main/java/org/elasticsearch/action/bulk/BulkProcessor.java

```diff
@@ -348,7 +348,7 @@ public void onFailure(Throwable e) {
                 });
                 success = true;
             } catch (InterruptedException e) {
-                Thread.interrupted();
+                Thread.currentThread().interrupt();
```

Comment:
In this case I would argue that

> rethrow InterruptedException whenever you do not know what code is higher up the call stack

translates to passing the exception to the listener - it's what we do for any other exception.